test/xml2json.test.js

```diff
@@ -405,6 +405,26 @@ describe("Error cases", function () {
     }
   });
 
+  it("unexpected text content", function () {
+    const xml = `<Edmx Version="4.0" xmlns="http://docs.oasis-open.org/odata/ns/edmx">X
+      <DataServices/>
+    </Edmx>`;
+    try {
+      csdl.xml2json(xml);
+      assert.fail("should not get here");
+    } catch (e) {
+      assert.strictEqual(
+        e.message.split("\n")[0],
+        "Unexpected text content in Edmx"
+      );
+      assert.deepStrictEqual(e.parser, {
+        construct: "<DataServices/",

```

Comment:
The "construct" in the enriched exception looks strange for errors in the text content.

Reply:
Yes, apparently the parser has already walked beyond the text, up to but not including the `>` of the next XML tag, so `line` and `column` are also off.